### PR TITLE
SoundWire: fix status handling

### PR DESCRIPTION
**Pull request by plbossart**

### CODE REVIEW

**Review by bardliao — DISMISSED** [review]

File: drivers/soundwire/bus.c [hunk 1]
```diff
@@ -1264,9 +1272,19 @@ static int sdw_initialize_slave(struct sdw_slave *slave)
 	if (slave->bus->prop.quirks & SDW_MASTER_QUIRKS_CLEAR_INITIAL_CLASH) {
 		/* Clear bus clash interrupt before enabling interrupt mask */
 		status = sdw_read_no_pm(slave, SDW_SCP_INT1);
```
Comment (lyakh) [inline drivers/soundwire/bus.c:1274]: there's one more unchecked call to `sdw_read_no_pm()` below at line 1275, and a `sdw_write_no_pm()` after it.

**Review by lyakh — CHANGES_REQUESTED** [review]

**Review by RanderWang — APPROVED** [review]

File: drivers/soundwire/bus.c (hunk 1 above)
Comment (plbossart) [reply drivers/soundwire/bus.c:1274]: nice catch, thanks for the review @lyakh

Comment (plbossart) [reply drivers/soundwire/bus.c:1274]: fixed now

**Review by kv2019i — COMMENTED** [review]
One more case for ret value checking?

File: drivers/soundwire/bus.c [hunk 2]
```diff
@@ -1264,15 +1272,30 @@ static int sdw_initialize_slave(struct sdw_slave *slave)
...
 		/* Clear parity interrupt before enabling interrupt mask */
 		status = sdw_read_no_pm(slave, SDW_SCP_INT1);
+		if (status < 0) {
+			dev_err(&slave->dev,
+				"SDW_SCP_INT1 read failed:%d\n", status);
+			return status;
+		}
 		if (status & SDW_SCP_INT1_PARITY) {
 			dev_warn(&slave->dev, "PARITY error detected before INT mask is enabled\n");
 			sdw_write_no_pm(slave, SDW_SCP_INT1, SDW_SCP_INT1_PARITY);
```
Comment (kv2019i) [inline drivers/soundwire/bus.c]: L1301, check ret value...?

**Review by kv2019i — COMMENTED** [review]
New try, see comment inline.

File: drivers/soundwire/bus.c (hunk 2 above)
Comment (plbossart) [reply drivers/soundwire/bus.c]: yes, I forgot to push this is a different commit.

File: drivers/soundwire/bus.c [hunk 3]
```diff
@@ -897,6 +901,10 @@ static int sdw_bus_wait_for_clk_prep_deprep(struct sdw_bus *bus, u16 dev_num)
 	do {
 		val = sdw_bread_no_pm(bus, dev_num, SDW_SCP_STAT) &
 			SDW_SCP_STAT_CLK_STP_NF;
+		if (val < 0) {
+			dev_err(bus->dev, "SDW_SCP_STAT bread failed:%d\n", val);
+			return val;
+		}
 		if (!val) {
 			dev_info(bus->dev, "clock stop prep/de-prep done slave:%d",
 				 dev_num);
```
Comment (lyakh) [inline drivers/soundwire/bus.c]: unrelated to this patch series: does this really deserve a `dev_info()`?

File: drivers/soundwire/bus.c [hunk 4]
```diff
@@ -1273,6 +1286,11 @@ static int sdw_initialize_slave(struct sdw_slave *slave)
 	    !(slave->prop.quirks & SDW_SLAVE_QUIRKS_INVALID_INITIAL_PARITY)) {
 		/* Clear parity interrupt before enabling interrupt mask */
 		status = sdw_read_no_pm(slave, SDW_SCP_INT1);
+		if (status < 0) {
+			dev_err(&slave->dev,
+				"SDW_SCP_INT1 read failed:%d\n", status);
```
Comment (lyakh) [inline drivers/soundwire/bus.c]: now we have two identical error messages at two different code locations... I find this confusing when debugging stuff, so, maybe better make them different?

File: drivers/soundwire/bus.c [hunk 5]
```diff
@@ -1293,7 +1298,13 @@ static int sdw_initialize_slave(struct sdw_slave *slave)
 		}
 		if (status & SDW_SCP_INT1_PARITY) {
 			dev_warn(&slave->dev, "PARITY error detected before INT mask is enabled\n");
-			sdw_write_no_pm(slave, SDW_SCP_INT1, SDW_SCP_INT1_PARITY);
+			ret = sdw_write_no_pm(slave, SDW_SCP_INT1, SDW_SCP_INT1_PARITY);
+			if (ret < 0) {
+				dev_err(&slave->dev,
+					"SDW_SCP_INT1 write failed:%d\n", ret);
```
Comment (lyakh) [inline drivers/soundwire/bus.c]: same with these two messages

**Review by lyakh — DISMISSED** [review]
some minor optional comments.

File: drivers/soundwire/bus.c (hunk 3 above)
Comment (plbossart) [reply drivers/soundwire/bus.c]: fixed in follow-up patch, thanks for the suggestion.

File: drivers/soundwire/bus.c (hunk 4 above)
Comment (plbossart) [reply drivers/soundwire/bus.c]: added unique messages in follow-up commit.

File: drivers/soundwire/bus.c (hunk 5 above)
Comment (plbossart) [reply drivers/soundwire/bus.c]: fixed.

**Review by lyakh — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

**Pull request merged**